algorithms/src/snark/marlin/ahp/indexer/indexer.rs

```diff
@@ -228,24 +236,20 @@ impl<F: PrimeField, MM: MarlinMode> AHPForR1CS<F, MM> {
 
     pub fn evaluate_index_polynomials<C: ConstraintSynthesizer<F>>(
         c: &C,
+        id: CircuitId,
         point: F,
     ) -> Result<impl Iterator<Item = F>, AHPError> {
         let state = Self::index_helper(c)?;
         let mut evals = [
-            ("a", state.a_evals, state.non_zero_a_domain),
-            ("b", state.b_evals, state.non_zero_b_domain),
-            ("c", state.c_evals, state.non_zero_c_domain),
+            (state.a_evals, state.non_zero_a_domain),
+            (state.b_evals, state.non_zero_b_domain),
+            (state.c_evals, state.non_zero_c_domain),
         ]
         .into_iter()
-        .flat_map(move |(matrix, evals, domain)| {
-            let labels = [
-                format!("row_{matrix}"),
-                format!("col_{matrix}"),
-                format!("val_{matrix}"),
-                format!("row_col_{matrix}"),
-            ];
+        .flat_map(move |(evals, domain)| {
+            let labels = Self::index_polynomial_labels(&["a", "b", "c"], vec![id]);
```

Comment:
```suggestion
            let labels = Self::index_polynomial_labels(&["a", "b", "c"], [id]);
```

Reply:
This creates various downstream issues, not sure if it's worth to pursue.

We're not allowed to just pass an array: `function arguments must have a statically known size, borrowed types always have a known size: `&``. If we pass a reference to the array, the array is a temporary value which can't be referenced outside of this function.

Given that id is cheap to copy, I would leave it as is

Reply:
The iterator approach you mention in the next comment is great